src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+    ) -> Path:
+        return self._raw_base_dir / self._get_file_name(
+            year=year, month=month, day=day
+        )
+
+    def _get_processed_path(self, raw_path: Path) -> Path:
+        return self._processed_base_dir / raw_path.parts[-1]
+
+    def _get_url(self, year: str, month: str, day: str) -> str:
+
+        # Convert month from month number (in string format) to
+        # three-letter name
+        month = calendar.month_abbr[int(month)]
+
+        if self._frequency == "daily":
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+                f".CHIRPS/.v2p0/.daily-improved/.global/."
+                f"{str(self._resolution).replace('.', 'p')}/.prcp/"
+                f"T/%28{day}%20{month}%20{year}%29%28{day}%20{month}%20{year}"
+                "%29RANGEEDGES/data.nc"
+            )
+        else:  # monthly
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+                ".CHIRPS/.v2p0/.monthly/.global/.precipitation/"
+                f"T/%28{month}%20{year}%29%28{month}%20{year}"
+                "%29RANGEEDGES/data.nc"
+            )
+        return url
+
+    def _get_last_available_date(self):
+        """Get the most recent date for which data is available."""
+        if self._frequency == "daily":
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/.CHIRPS/.v2p0/"
+                ".daily-improved/.global/"
+                f"{str(self._resolution).replace('.', 'p')}/"
+                ".T/last/subgrid/0./add/T/table%3A/1/%3Atable/.csv"
+            )
+
+            datetime_object = datetime.strptime(
+                pd.read_csv(url).values[0][0], "%d %b %Y"
+            )
+
+        else:  # monthly
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/.CHIRPS/.v2p0/"
+                ".monthly/.global/"
+                ".T/last/subgrid/0./add/T/table%3A/1/%3Atable/.csv"
+            )
+
+            datetime_object = datetime.strptime(
+                pd.read_csv(url).values[0][0], "%b %Y"
+            )
+            day = calendar.monthrange(
+                datetime_object.year, datetime_object.month
+            )[1]
+            datetime_object = datetime_object.replace(day=day)
+
+        return datetime_object.date()
+
+
+class ChirpsMonthly(Chirps):
+    """
+    .
+
+    Examples
+    --------
+    >>> from aatoolbox import create_country_config, CodAB, ChirpsMonthly
+    >>> country_config = create_country_config(iso3="bfa")
+    >>> codab = CodAB(country_config=country_config)
+    >>> codab.download()
+    >>> admin0 = codab.load(admin_level=0)
+    >>> geo_bounding_box = GeoBoundingBox.from_shape(admin0)
+    >>> chirps_monthly = ChirpsMonthly(country_config=country_config,
+                                    geo_bounding_box=geo_bounding_box)
+    >>> chirps_monthly.download(
+            start_year=2007,
```

Comment:
you should add 3 dots if you start a new line in the examples. Else the automatically built documentation shows them as grey